code-challenges.test.js

```diff
@@ -14,54 +14,182 @@
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fibLength1 = 6
-// Expected output: [1, 1, 2, 3, 5, 8]
+describe("fibFunction", () => {
+  it(`Takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence.`, () => {
+    const fibLength1 = 6;
+    const fibLength2 = 10;
 
-const fibLength2 = 10
-// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+    expect(fibFunction(fibLength1)).toEqual([1, 1, 2, 3, 5, 8]);
+    expect(fibFunction(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55]);
+    expect(fibFunction(0)).toEqual(
+      "Please enter a number that's at least 2, or higher."
+    );
+  });
+});
+
+// GOOD FAIL: fibfunction is not defined
 
 // b) Create the function that makes the test pass.
+/*
+--------- FUNCTION INFORMATION:
+Describe what the function will do: Function will take in a number and return an array of numbers in the fibonacci sequence that matches the length of the number.
+Function Name: fibFunction
+Function Parameter(s): 1, number
+--------- INPUT / OUTPUT:
+input 1: number
+output 1: array of numbers in fibonacci sequence
+--------- STEPS/METHODS:
+1. I'll declare an array to store the first two values of the fibonacci sequence, as well as an array with [1] to start us off.
+2. I'll use a conditional statement to ensure the number being entered is greater than 2. If it isn't, I'll return a string notifying that the number needs to be at least 2 or higher.
+3. I'll use a for loop to iterate through the fibonacci sequence, and return the corresponding output to an array of the specified length.
+4. Return the resulting array.
+*/
+
+// The Fibonacci sequence is the integer sequence where the first two terms are 0 and 1. After that, the next term is defined as the sum of the previous two terms.
+
+const fibFunction = (n) => {
+  // The first two terms in the fibonacci sequence are 0 and 1, so I'm creating an array to store those values.
+  let fib = [0, 1];
+  // Starting the array that the sequence will be stored in with a 1.
+  let newFib = [1];
+
+  if (n > 0) {
+    // Looping to populate an array that will be the length of the specified input.
+
+    for (let i = 2; i <= n; i++) {
+      // Current index = (index - 2) + (index -1)
+      // Starts at index 2, so this will start iterating with 0 + 1, then 1 + 1, 2 + 1, etc.
+      fib[i] = fib[i - 2] + fib[i - 1];
+      // Push the added total to the current index of newFib.
+      newFib.push(fib[i]);
+    }
+    // Return the total populated array.
+    return newFib;
+  } else if (n < 2) {
+    // If number entered is less than 2, return this string:
+    return "Please enter a number that's at least 2, or higher.";
+  } else {
+    // Edge case, because you never know. (:
+    return "Oops! Something went wrong.";
+  }
+};
```

Comment:
Awesome work!  For that last edge case, you could add a test and pass a string to test it works.  Really like that you are thinking about these things though!